youtube_transcript_api/formatters.py

```diff
@@ -0,0 +1,174 @@
+from collections import defaultdict
+import json
+import re
+
+
+def parse_timecode(time):
+    """Converts a `time` into a formatted transcript timecode.
+
+    :param time: a float representing time in seconds.
+    :type time: float
+    :return: a string formatted as a timecode, 'HH:MM:SS,MS'
+    :rtype str
+
+    :example:
+    >>> parse_timecode(6.93)
+    '00:00:06,930'
+    """
+    
+    time = float(time)
+    times = {
+        'hours': str(int(time) // 3600).rjust(2, '0'),
+        'mins': str(int(time) // 60).rjust(2, '0'),
+        'secs': str(int(time) % 60).rjust(2, '0'),
+        'ms': str(int(round((time - int(time))*1000, 2))).rjust(3, '0')
+    }
+    return "{hours}:{mins}:{secs},{ms}".format(**times)
+
+
+class TranscriptFormatter(object):
+    """Abstract Base TranscriptFormatter class
+
+    This class should be inherited from to create additional
+     custom transcript formatters.
+    """
+    HTML_TAG_REGEX = re.compile(r'<[^>]*>', re.IGNORECASE)
+    DELIMITER = ''
+
+    @classmethod
+    def combine(cls, transcripts):
```

Comment:
I agree. I was actually rethinking formatters the other day before writing tests. Rethinking the same thing, a formatter as an instance might be much more flexible/powerful. Thinking about them in similar terms of how a Django REST Framework Serializer works. 👍   How you can pass a Model instance into a Serializer and it can translate that into a given format.

Reply:
For sure, we could implement this using a more generic approach like DRF serializers, but I feel like that would be a bit over-engineered, for this project.